filebeat/harvester/reader/multiline.go

```diff
@@ -215,12 +216,18 @@ func (mlr *Multiline) resetBuffer(m Message) {
 	mlr.addLine(m)
 	// Timestamp of first message is taken as overall timestamp
 	mlr.ts = m.Ts
-	mlr.fields = m.Fields
+	if m.Fields != nil {
+		mlr.fields = m.Fields
+	} else {
+		mlr.fields = common.MapStr{}
+	}
```

Comment:
Yeah, that should simplify the code. Thinking of doing this in an other PR first.